docs/modules/ROOT/pages/cli_reference.adoc

```diff
@@ -0,0 +1,61 @@
+= CLI Reference
+
+In this section you will learn about:
+
+* [x] ways to configure `ike` (through flags, files et al)
+* [x] complete list of commands and their available flags
+
+[#configuration]
+== Configuration
+
+There are several ways to configure `ike` commands:
+
+. Flags
+. Environment variables
+. Configuration file
+
+Each item takes precedence over the item below it.
+
+Environment variables are prefixed with `IKE` and have fully qualified names, for example
+in case of `develop` command and its `port` flag corresponding environment variable is
+`IKE_DEVELOP_PORT`.
+
+Each of the commands can be persisted in the configuration file with all their flags respectively.
+
+In case of `develop` command `yaml` file could look like:
+
+[source,yml]
+----
+develop:
+  deployment: test
+  run: "java -jar config.jar"
+  port: 9876
+----
+
+See <<ike,ike root command>> to learn about the global config flag and available formats.
+
+Below you can find a documented list of all commands and their available flags.
+
+== Available commands
+
+[#ike]
+=== `ike`
+
+Main command, which starts the operator.
+// TODO brief mentionn about operator and how it works + link to the dedicated docs
```

Comment:
@aslakknutsen you can use

```
 ```suggestion
 \```
```

block to fill-in this gap. But this equally well can go with the separated PR.